expression_json.go

```diff
@@ -31,71 +31,77 @@ func parseJsonField(data []interface{}) (Expression, error) {
 	return out, nil
 }
 
-func parseJsonEquals(data []interface{}) (Expression, error) {
+func parseJsonComparison(data []interface{}) (Expression, Expression, error) {
 	lhsData, ok := data[1].([]interface{})
 	if !ok {
-		return nil, errors.New("invalid equals expression lhs format")
+		return nil, nil, errors.New("invalid equals expression lhs format")
 	}
 
 	rhsData, ok := data[2].([]interface{})
 	if !ok {
-		return nil, errors.New("invalid equals expression rhs format")
+		return nil, nil, errors.New("invalid equals expression rhs format")
```

Comment:
this function is used by more than just equals (i.e. not equals). So this error may not reflect the accurate caller.